mp4e/mp4.go

```diff
@@ -0,0 +1,308 @@
+package mp4e
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"sort"
+	"strings"
+
+	"github.com/Eyevinn/mp4ff/avc"
+	"github.com/Eyevinn/mp4ff/mp4"
+
+	"github.com/eluv-io/errors-go"
+)
+
+type Mp4Info struct {
+	Errors            []string      // problems found
+	FragmentCount     uint64        // total number of fragments
+	InitInfo          *InitInfo     // moov
+	SampleCount       uint64        // total number of samples/frames
+	SampleCountMax    uint64        // most number of samples in a segment
+	SampleCountMin    uint64        // least number of samples in a segment
+	SampleDurationMax uint64        // longest sample duration
+	SampleDurationMin uint64        // shortest sample duration
+	Segments          []SegmentInfo //
+	Size              uint64        // file size
+	Timescale         uint32        //
+}
+
+func (s *Mp4Info) AddError(err string, seg, frag int, pts int64) {
+	seg++
```

Comment:
This is to make them 1-indexed, right?

Reply:
Hmm yeah, I should probably make a comment about it. If this is ever used elsewhere it would be better to require the +1 before calling this function